external-crates/move/crates/move-compiler/src/expansion/ast.rs

```diff
@@ -99,6 +99,18 @@ pub type Attributes = UniqueMap<Spanned<KnownAttribute>, Attribute>;
 // Modules
 //**************************************************************************************************
 
+#[derive(Debug, Clone, Copy)]
+/// Specifies a source target or dependency
+pub enum TargetKind {
+    /// A source module
+    Source,
+    /// A dependency where bytecode is being generated
+    DependencyBeingCompiled,
+    /// A dependency only used for linking.
+    /// No bytecode or CompiledModules are generated
+    DependencyBeingLinked,
```

Comment:
I kind of hate these names, open to suggestions

Reply:
I agree, these names don't communicate what is going on very well.

A few suggestions:

1) 
```
pub enum TargetKind {
  Dependency { from_source bool },
  RootProject // or similar
}
```

2) 
```
pub enum TargetKind {
  SourceDependency,
  BytecodeDependency,
  Source
}
```

I also asked ChatGPT:

```
enum PackageType {
    MainProject,
    SourceDependency,
    BytecodeDependency,
}
```

Reply:
My issue with "BytecodeDependency" is it makes it feel like it was from an `.mv` file